jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletPathMapping.java

```diff
@@ -112,10 +112,10 @@ public ServletPathMapping(PathSpec pathSpec, String servletName, String pathInCo
             case PREFIX_GLOB:
                 _mappingMatch = MappingMatch.PATH;
                 _servletPath = pathSpec.getPrefix();
-                _matchValue = _servletPath.startsWith("/") ? _servletPath.substring(1) : _servletPath;
                 _pathInfo = matchedPath != null
                     ? matchedPath.getPathInfo()
                     : _servletPath.length() == pathInContext.length() ? null : pathInContext.substring(_servletPath.length());
+                _matchValue = _pathInfo == null ? "" : _pathInfo.startsWith("/") ? _pathInfo.substring(1) : _pathInfo;
```

Comment:
Why is this using `_pathInfo` and not `pathInContext`?

Reply:
The spec requires that for in a context "/ctx", with a pattern "/foo/*" then for a request to "/ctx/foo/info", the `contextPath` will be "/ctx", the `servletPath` will be "/foo" and the `pathInfo` "/info".   We had previously implemented that the match value was based on the `servletPath` (so "foo"), but it needs to be based on the part that matched the *, i.e. the `pathInfo` without the leading slash (so  "info").

The path in context  is "/foo/info", so that is `servletPath` + `pathInfo` and is not what we want for the matchValue.